### ISSUE TITLE
split_all_top_level_comma_separated_values

### ISSUE THREAD

**Issue by mpanarin** [google/yapf#745]
`split_all_top_level_comma_separated_values` and `dedent_closing_brackets` are not working correctly together.
code to reproduce:
```python
def test():
    return {
        "foo": "bar",
        "spam": "eggs",
    }
```
result:
```python
def test():
    return {
        "foo": "bar",
        "spam": "eggs", }
```
expected result: no changes to codeblock
config used:
```ini
[style]
based_on_style = pep8
split_all_top_level_comma_separated_values = true
allow_split_before_dict_value = false
continuation_indent_width = 4
dedent_closing_brackets = true
coalesce_brackets = true
split_before_dot = false
each_dict_entry_on_separate_line = true
indent_dictionary_value = true
spaces_before_comment = 2
column_limit = 120
```

**Comment by sergiogiro** [comment]
Working on a fix for this. Should be ready in a couple weeks

**Comment by sergiogiro** [comment]
Mayber this is fixed by https://github.com/google/yapf/commit/405a524e0a8a6923805bc23ce4acae7ea2102c1e
already?
(Seems the same as
https://github.com/google/yapf/issues/771
)

**Comment by alexpersin** [comment]
I experienced the same issue and can confirm it was fixed by upgrading from 0.28.0 to 0.29.0
